tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3775Test.php

```diff
@@ -0,0 +1,376 @@
+        $pageRelation->setPageCreator($pageTwo);
+        $pageRelation->setPageReferenced($pageThree);
+        $manager->persist($pageRelation);
+
+        $pageRelation = new DDC3775PageRelation();
+        $pageRelation->setPageCreator($page);
+        $pageRelation->setPageReferenced($pageThree);
+        $manager->persist($pageRelation);
+
+        $manager->flush();
+        $manager->clear();
+    }
+
+    public function testRelationsPagesShouldBeProxyIfNotPreviouslyFetched()
+    {
+        // After setupDoctrine, pages in $pageIdsToFetch have been added to the identitymap.
+        $pageIdsToFetch = [1];
+        $this->setupDoctrineCaches($pageIdsToFetch);
+
+        $qb = $this->_em->getRepository(__NAMESPACE__ . '\DDC3775PageRelation')->createQueryBuilder('r')
+            ->select('r');
+
+        $relations = $qb->getQuery()->getResult();
+
+        foreach ($relations as $relation) {
+            // This page was not fetched before so it should be a proxy
+            if (false === in_array($relation->getPageCreator()->getId(), $pageIdsToFetch)) {
+                $this->assertInstanceOf('Doctrine\ORM\Proxy\Proxy', $relation->getPageCreator());
+            } elseif (false === in_array($relation->getPageReferenced()->getId(), $pageIdsToFetch)) {
+                $this->assertInstanceOf('Doctrine\ORM\Proxy\Proxy', $relation->getPageReferenced());
+            }
+        }
+    }
+
+
+    public function testRelationsPagesShouldHaveNonNullTitleBecauseItIsNonNullable() {
+        // After setupDoctrine, pages in $pageIdsToFetch have been added to the identitymap.
+        $pageIdsToFetch = [1];
+        $this->setupDoctrineCaches($pageIdsToFetch);
+
+        $qb = $this->_em->getRepository(__NAMESPACE__ . '\DDC3775PageRelation')->createQueryBuilder('r')
+            ->select('r, p1, p2')
+            ->leftJoin('r.pageCreator', 'p1')
+            ->leftJoin('r.pageReferenced', 'p2');
+
+        $relations = $qb->getQuery()->getResult();
+
+        foreach ($relations as $relation) {
+            $this->assertNotNull($relation->getPageCreator()->getTitle());
+            $this->assertNotNull($relation->getPageReferenced()->getTitle());
+        }
+    }
+
+    protected function setupDoctrineCaches($preFetchPages = [1]) {
+        $qb = $this->_em->getRepository(__NAMESPACE__ . '\DDC3775Page')->createQueryBuilder('p')
+            ->select('p')
+            ->where('p.id IN (:ids)')
+            ->setParameter('ids', $preFetchPages);
+
+        $allPages = $qb->getQuery()->getResult();
+
+        $entityIds = [];
+        foreach($allPages as $user) {
+            $entityIds = $user->getId();
+        }
+
+        $qb = $this->_em->getRepository(__NAMESPACE__ . '\DDC3775Page')->createQueryBuilder('p')
+            ->select('partial p.{id}, pm')
+            ->leftJoin('p.pageMembers', 'pm', 'WITH', 'p.id IN (:ids)')
+            ->setParameter('ids', $entityIds);
+
+        $qb->getQuery()->getResult();
```

Comment:
So what is happening here (theory) is that the partial hydration hint is leaking in the hydration of the association `Page` objects, and producing real instances rather than proxies.

`DDC3775Page#2` and `DDC3775Page#3` are not supposed to be loaded in the query, but the `UnitOfWork` will allocate proxy objects for them anyway during the hydration of `DDC3775PageMember` objects. During that operation, deciding whether a proxy or a partial object are required is broken.

That's our bug to track down.